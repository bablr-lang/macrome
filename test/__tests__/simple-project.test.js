const { resolve } = require('path');
const { testProject } = require('./test-project');
const { gitStatus, isClean } = require('./utils');

const { writeFile, readFile, unlink } = require('fs/promises');

const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

describe('simple project', () => {
  const test = testProject(resolve(__dirname, '../sandbox/projects/simple-project'));

  describe('watch', () => {
    test.watchSetup();

    it('simple-project.js', async () => {
      const filePath = 'lib/simple-project.js';
      const originalContent = await readFile(filePath, 'utf8');

      await unlink(filePath);
      await sleep(100);

      expect(gitStatus()).toMatchSnapshot();

      await writeFile(filePath, originalContent + '\n');
      await sleep(100);

      expect(gitStatus()).toMatchSnapshot();

      await writeFile(filePath, originalContent);
      await sleep(100);

      expect(isClean('.')).toBe(true);
    });
  });
});
