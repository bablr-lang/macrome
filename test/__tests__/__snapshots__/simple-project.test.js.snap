// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple project cleans 1`] = `
Array [
  " D lib/generated-simple-project.js",
]
`;

exports[`simple project watch simple-project.js 1`] = `
Array [
  " D lib/generated-simple-project.js",
  " D lib/simple-project.js",
]
`;

exports[`simple project watch simple-project.js 2`] = `
Array [
  " M lib/generated-simple-project.js",
  " M lib/simple-project.js",
]
`;
